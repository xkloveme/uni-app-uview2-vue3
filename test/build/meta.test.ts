import { PageRETargets, MetaRETargets, MetaParserTargets } from './meta.targets'
import { defaultPagesRE, defaultMetaRE, MetaParser } from '@root/build/vite-plugin-uni-meta'
describe('页面元数据', () => {
  test('页面路径匹配', async () => {
    const pathOutput = PageRETargets.map(
      path => (path + ': ').padEnd(45) + (defaultPagesRE.test(path) ? '命中页面' : '忽略'),
    )
    expect(pathOutput).toMatchSnapshot()
  })

  test('meta标签匹配', async () => {
    const metaOutput = MetaRETargets.map(str => str.match(defaultMetaRE)[0])
    expect(metaOutput).toMatchSnapshot()
  })

  test('meta标签解析', async () => {
    let alias = {
      微信: 'mp-weixin',
      app: 'app-plus',

      title: 'navigationBarTitleText',
    }
    let preset = {
      hide: {
        navigationStyle: 'custom',
      },
    }

    let metaWithoutAlias = MetaParser('<meta navigationBarTitleText="foo"></meta>', alias, preset)
    let metaWithAlias = MetaParser('<meta title="foo"></meta>', alias, preset)
    assert.deepStrictEqual(metaWithoutAlias, metaWithAlias, '别名应当生效')

    let metaWithPreset = MetaParser('<meta hide title="foo"></meta>', alias, preset)
    assert(metaWithPreset['navigationStyle'] === 'custom', '预设应当生效')

    const metaOutput = MetaParserTargets.map(str => MetaParser(str, alias, preset))
    expect(metaOutput).toMatchSnapshot()
  })
})
