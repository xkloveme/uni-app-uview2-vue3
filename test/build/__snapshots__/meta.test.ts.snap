// Vitest Snapshot v1

exports[`页面元数据 > meta标签匹配 1`] = `
[
  "<meta title=\\"foo\\" />",
  "<meta title=\\"foo\\"></meta>",
  "<meta title=\\"foo\\">
  </meta>",
  "<meta title=\\"foo\\" />",
  "<meta title=\\"foo\\"></meta>",
]
`;

exports[`页面元数据 > meta标签解析 1`] = `
[
  {
    "navigationBarTitleText": "foo",
  },
  {
    "navigationBarTitleText": "foo",
  },
  {
    "navigationBarTitleText": "foo",
  },
  {
    "mp-weixin": {
      "navigationBarTitleText": "foo",
    },
  },
  {
    "mp-weixin": {
      "navigationBarTitleText": "foo",
    },
  },
  {
    "navigationBarTitleText": "foo",
    "navigationStyle": "custom",
  },
]
`;

exports[`页面元数据 > 页面路径匹配 1`] = `
[
  "src/pages/index/index.vue:                   命中页面",
  "src/pages/foo/index.vue:                     命中页面",
  "src/pages/foo/bar/index.vue:                 命中页面",
  "src/pages/foo/components/bar.vue:            忽略",
  "src/pages/foo/bar/components/baz.vue:        忽略",
  "src/pages/foo/bar/component/baz.vue:         忽略",
  "src/pages/foo/static/bar.vue:                忽略",
  "src/pages/foo/bar/static/baz.vue:            忽略",
  "src/pages/static/foo.vue:                    命中页面",
  "src/pages/component/foo.vue:                 命中页面",
  "src/pages/components/foo.vue:                命中页面",
  "src/pages/components/components/foo.vue:     忽略",
]
`;
